examples/example_mnist_prune.py

```diff
@@ -0,0 +1,227 @@
+                        for weight in prunable_weights
+                    ])))
+    print("\n")
+
+
+def train_and_save(model, x_train, y_train, x_test, y_test):
+    model.compile(
+        loss="categorical_crossentropy",
+        optimizer="adam",
+        metrics=["accuracy"])
+
+    # Print the model summary.
+    model.summary()
+
+    # Add a pruning step callback to peg the pruning step to the optimizer's
+    # step. Also add a callback to add pruning summaries to tensorboard
+    callbacks = [
+        pruning_callbacks.UpdatePruningStep(),
+        #pruning_callbacks.PruningSummaries(log_dir=tempfile.mkdtemp())
+        pruning_callbacks.PruningSummaries(log_dir="/tmp/mnist_prune")
+    ]
+
+    model.fit(
+        x_train,
+        y_train,
+        batch_size=batch_size,
+        epochs=epochs,
+        verbose=1,
+        callbacks=callbacks,
+        validation_data=(x_test, y_test))
+    score = model.evaluate(x_test, y_test, verbose=0)
+    print("Test loss:", score[0])
+    print("Test accuracy:", score[1])
+
+    print_model_sparsity(model)
+
+    # Export and import the model. Check that accuracy persists.
+    _, keras_file = tempfile.mkstemp(".h5")
+    print("Saving model to: ", keras_file)
+    save_model(model, keras_file)
+    
+    print("Reloading model")
+    with prune.prune_scope():
+        loaded_model = load_qmodel(keras_file)
+    score = loaded_model.evaluate(x_test, y_test, verbose=0)
+    print("Test loss:", score[0])
+    print("Test accuracy:", score[1])
+
+
+def main():
+    # input image dimensions
+    img_rows, img_cols = 28, 28
+
+    # the data, shuffled and split between train and test sets
+    (x_train, y_train), (x_test, y_test) = mnist.load_data()
+
+    if K.image_data_format() == "channels_first":
+      x_train = x_train.reshape(x_train.shape[0], 1, img_rows, img_cols)
+      x_test = x_test.reshape(x_test.shape[0], 1, img_rows, img_cols)
+      input_shape = (1, img_rows, img_cols)
+    else:
+      x_train = x_train.reshape(x_train.shape[0], img_rows, img_cols, 1)
+      x_test = x_test.reshape(x_test.shape[0], img_rows, img_cols, 1)
+      input_shape = (img_rows, img_cols, 1)
+
+    x_train = x_train.astype("float32")
+    x_test = x_test.astype("float32")
+    x_train /= 255
+    x_test /= 255
+    print("x_train shape:", x_train.shape)
+    print(x_train.shape[0], "train samples")
+    print(x_test.shape[0], "test samples")
+
+    # convert class vectors to binary class matrices
+    y_train = to_categorical(y_train, num_classes)
+    y_test = to_categorical(y_test, num_classes)
+
+    pruning_params = {
+        "pruning_schedule":
+            pruning_schedule.ConstantSparsity(0.75, begin_step=2000, frequency=100)
+    }
+    
+    if prune_whole_model:
+        model = build_model(input_shape)
+        model = prune.prune_low_magnitude(model, **pruning_params)
+    else:
+        model = build_layerwise_model(input_shape, **pruning_params)
```

Comment:
could you add a branch for non-pruned model. And print the different sparsity. That would be very interesting!

Reply:
The non-pruned model will not be sparse (i.e., will have sparsity eq. 0). The printed sparsity just confirms that the model is pruned as desired (to 0.75).